### ISSUE TITLE
can you code to do classification for F1 metric (unbalanced data classification)?

### ISSUE THREAD

**Issue by Sandy4321** [ferryjul/fairCORELS#4]
you have something related to unbalanced data classification like 
class ConfusionMatrix(namedtuple('ConfusionMatrix', 'minority majority label truth')):
https://github.com/ferryjul/fairCORELS/blob/master/examples/metrics.py

so
can you code to do classification for F1 metric (unbalanced data classification)?

**Comment by ferryjul** [comment]
Hello, thank you for your interest in FairCORELS, and sorry for the late answer!

As you mention, additional tools can be imported from FairCORELS: ConfusionMatrix and Metric objects. 
They are designed specifically to measure statistical fairness. 
For more details about this concept, please refer to the following paper: https://fairware.cs.umass.edu/papers/Verma.pdf.

In order to facilitate its use, FairCORELS is designed in a scikit-learn-like fashion. I would hence recommend using sklearn existing methods to measure f1-score, e.g., by doing:
score = sklearn.metrics.f1_score(y_true, y_pred) 
where y_true is the vector of true labels and y_pred the vector of predicted labels, which can easily be computed for input set X by any FairCORELSClassifier object clf  with 'y_pred = clf.predict(X)'.

Concerning the use of the F1-score as an objective within FairCORELS, it would be possible to integrate it (modifying file ["corels.cpp").](https://github.com/ferryjul/fairCORELS/blob/master/faircorels/src/corels/src/corels.cpp) However, it would necessitate modifying objective bounds.
